test/parallel/test-http2-compat-expect-continue-check.js

```diff
@@ -0,0 +1,78 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const testResBody = 'other stuff!\n';
+
+// Checks the full 100-continue flow from client sending 'expect: 100-continue'
+// through server receiving it, triggering 'checkContinue' custom handler,
+// writing the rest of the request to finally the client receiving to.
+
+function handler(req, res) {
+  console.error('Server sent full response');
+
+  res.writeHead(200, {
+    'content-type': 'text/plain',
+    'abcd': '1'
+  });
+  res.end(testResBody);
+}
+
+const server = http2.createServer(
+  common.mustNotCall('Full request received before 100 Continue')
+);
+
+server.on('checkContinue', common.mustCall((req, res) => {
+  console.error('Server received Expect: 100-continue');
+
+  res.writeContinue();
+
+  // timeout so that we allow the client to receive continue first
+  setTimeout(common.mustCall(() => handler(req, res)), 100);
```

Comment:
If we are keeping the timeout, we should multiply `100` with `common.platformTimeout`.